docs/howto/how_to_build_and_debug.md

```diff
@@ -0,0 +1,273 @@
+# How to Build and Debug
+
+* [1. Install Docker](#install-docker)
+* [2. Build Apollo](#build-apollo)
+* [3. Lint Apollo](#lint-apollo)
+* [4. Test Apollo](#test-apollo)
+* [5. Possible errors and solution](#possible-errors-and-solution)
+* [6. Debugging Apollo](#debugging-apollo)
+* [7. Remote Debugging](#remote-debugging)
+* [8. Legal Disclaimer](#legal-disclaimer)
+
+## Install Docker
+
+Please follow the [Apollo Software Installation Guide](https://github.com/ApolloAuto/apollo/blob/master/docs/quickstart/apollo_software_installation_guide.md#Set-up-the-Docker-environment).
+
+## Build Apollo
+
+### Start container
+We provide a build x86 image named dev-x86_64-18.04-*. The Container will mount your local apollo repo to /apollo.
+```bash
+bash docker/scripts/dev_start.sh
+```
+### Get into the container
+```bash
+bash docker/scripts/dev_into.sh
+```
+### Build modules
+```bash
+bash apollo.sh build -option
+```
+if you want to build apollo in diversified modules, you can choose 'build_gdb','build_opt' etc.
```

Comment:
build modules more introduce could add link here

Reply:
Try to split the PR into separate ones,  one for VSCode removal.
And, please don't cover so many topics in one markdown. Try little ones such as 'options for apollo build', 'how to run unit test in Apollo' and 'how to run apollo coverage', 'Linters and formatters in Apollo', 'How to use gdb server to debug apollo modules', etc. 'how to pull apollo docker image (CN users only)', 'how to build apollo docker images', etc.